### PR TITLE
[AIRFLOW-2357] Add persistent volume for the logs

### PR DESCRIPTION
**Pull request by Fokko**
The logs are kept inside of the worker pod. By attaching a persistent disk we keep the logs and make them available for the webserver.

Right now, when a pod exits, the log is lost forever because it is still in the container. By mounting a persistent volume we can easily fix this and this allows us to have logs on our local machines (minikube). In production you might want to write your logs to GCS/S3/etc.

Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2357
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-2357\], code changes always need a JIRA issue.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: airflow/contrib/executors/kubernetes_executor.py [hunk 1]
```diff
@@ -556,15 +562,17 @@ def start(self):
...
                 key, command, executor_config
             ))
         kube_executor_config = KubernetesExecutorConfig.from_dict(executor_config)
         self.task_queue.put((key, command, kube_executor_config))
 
     def sync(self):
-        self.log.info("self.running: {}".format(self.running))
-        self.log.info("self.queued: {}".format(self.queued_tasks))
+        if self.running:
+            self.log.info("self.running: {}".format(self.running))
```
Comment (bolkedebruin) [inline airflow/contrib/executors/kubernetes_executor.py]: can you use the new format if we are updating anyway?

File: docs/kubernetes.rst [hunk 2]
```diff
@@ -1,3 +1,13 @@
+Kubernetes Executor
+===================
+
+The kubernetes executor is introduced in Apache Airflow 1.10.0. The Kubernetes executor will create a new pod for every task instance.
+
+Example helm charts are available at `scripts/ci/kubernetes/kube/{airflow,volumes,postgres}.yaml`. The volumes are optional and depend on your configuration. There are two volumes available:
```
Comment (bolkedebruin) [inline docs/kubernetes.rst]: We only incude scripts/ci in the _source_ distribution. I think it is nice to the use to note that.

**Review by bolkedebruin — CHANGES_REQUESTED** [review]
Small nits. Otherwise LGTM

File: airflow/contrib/executors/kubernetes_executor.py (hunk 1 above)
Comment (bolkedebruin) [reply airflow/contrib/executors/kubernetes_executor.py]: (everywhere :P )

File: docs/kubernetes.rst (hunk 2 above)
Comment (Fokko) [reply docs/kubernetes.rst]: Ack

File: airflow/contrib/executors/kubernetes_executor.py (hunk 1 above)
Comment (Fokko) [reply airflow/contrib/executors/kubernetes_executor.py]: Yes

**Pull request closed without merging**